web/app/scripts/state/filters-service.js

```diff
@@ -16,29 +23,45 @@
          *
          * @param {Object} filters FilterBar's filters object to restore on load.
          */
-        svc.saveFilters = function(filters) {
-            localStorageService.set(storageName, JSON.stringify(filters));
-        };
+        function saveFilters(filters) {
+            localStorageService.set(storageName, filters);
+        }
+
+
+        /**
+         * A simple method for returning only the relevant portion of oldFilters
+         */
+        function get(key) {
+            return oldFilters[key] || {};
+        }
+
+
+        function clear() {
+            localStorageService.remove(storageName);
+        }
+
 
         /**
          * Broadcast event to trigger setting previously stored filters back on the filter bar.
          */
-        svc.restoreFilters = function() {
+        function restoreFilters() {
             var filterString = localStorageService.get(storageName);
-            var filterObj = !!filterString ? JSON.parse(filterString) : {};
+            var filterObj = !!filterString ? filterString : {};
```

Comment:
The localStorageService already parses in its get method, so I think this was noop at best